src/makeCalculator.js

```diff
@@ -4,7 +4,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    operate(action, number) {
+      this.result = action(this.result, number);
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(currentResult, number) {
+      return currentResult + number;
```

Comment:
Instead of returning a new value, use the '+=' operator to add directly to 'currentResult'.